Python/import.c

```diff
@@ -2363,9 +2363,14 @@ get_path_importer(PyThreadState *tstate, PyObject *path_importer_cache,
     PyObject *importer;
     Py_ssize_t j, nhooks;
 
-    /* These conditions are the caller's responsibility: */
-    assert(PyList_Check(path_hooks));
-    assert(PyDict_Check(path_importer_cache));
+    if (!PyList_Check(path_hooks)) {
+        PyErr_SetString(PyExc_RuntimeError, "sys.path_hooks is not a list");
+        return NULL;
+    }
+    if (!PyDict_Check(path_importer_cache)) {
```

Comment:
You might replace PyDict_GetItemWithError() with PyDict_GetItemRef() below to avoid _PyErr_Occurred() call.

Reply:
I am going to backport this change.

BTW, I have a separate branch in which most of `PyDict_GetItemWithError()` are replaced with `PyDict_GetItemRef()`. I will make a PR or several PRs from it in future.

Reply:
> I am going to backport this change.

Oh right. In this case, just ignore my remark. It's more a "coding style" change than a fix, the current code is correct.